src/main/java/com/gmail/stepura/volodymyr/service/implementation/CsvToBeanParserImpl.java

```diff
@@ -0,0 +1,108 @@
+import com.gmail.stepura.volodymyr.model.User;
+import com.gmail.stepura.volodymyr.service.CsvToBeanParserService;
+import com.gmail.stepura.volodymyr.service.ProductService;
+import com.gmail.stepura.volodymyr.service.ReviewService;
+import com.gmail.stepura.volodymyr.service.RoleService;
+import com.gmail.stepura.volodymyr.service.UserService;
+import com.univocity.parsers.common.record.Record;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.extern.log4j.Log4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Log4j
+@Service
+public class CsvToBeanParserImpl implements CsvToBeanParserService {
+    public static final String USER_ID = "UserId";
+    public static final String USER = "USER";
+    public static final String PRODUCT_ID = "ProductId";
+    public static final String PROFILE_NAME = "ProfileName";
+    public static final String ID = "Id";
+    public static final String HELPFULNESS_NUMERATOR = "HelpfulnessNumerator";
+    public static final String HELPFULNESS_DENOMINATOR = "HelpfulnessDenominator";
+    public static final String SCORE = "Score";
+    public static final String TIME = "Time";
+    public static final String SUMMARY = "Summary";
+    public static final String TEXT = "Text";
+    public static final String PASSWORD = "1111";
+    private final ReviewService reviewService;
+    private final UserService userService;
+    private final ProductService productService;
+    private final RoleService roleService;
+
+    @Autowired
+    public CsvToBeanParserImpl(ReviewService reviewService,
+                               UserService userService,
+                               ProductService productService,
+                               RoleService roleService) {
+        this.reviewService = reviewService;
+        this.userService = userService;
+        this.productService = productService;
+        this.roleService = roleService;
+    }
+
+    @Override
+    public void saveBeans(List<Record> records) {
+        log.info("Starting to parse List<Record>");
+        Role roleUser = roleService.getRoleByRoleName(USER);
+        Set<User> userHashSet = new HashSet<>();
+        Set<Product> productHashSet = new HashSet<>();
+
+        for (Record row : records) {
+            userHashSet.add(parseUser(row, roleUser));
+            productHashSet.add(parseProduct(row));
+        }
+
+        userService.saveAll(userHashSet);
+        log.info("Saving of the users are successful!");
+
+        productService.saveAll(productHashSet);
+        log.info("Saving of the products are successful!");
+
+        reviewService.saveAll(records.stream()
+                .map(this::parseReview)
+                .collect(Collectors.toList()));
+        log.info("Saving of the reviews are successful!");
+    }
+
+    private User parseUser(Record row, Role role) {
```

Comment:
it will be more logical to fetch the role from db in this method

Reply:
I did this hardcoding, because of luck in performance, when we had the big one file for parsing. I even had the idea to add the multithreading for parsing.